File: frontend/src/app/services/services/room-types.service.ts
import { Injectable } from '@angular/core';
import { ApiClient } from '../../core/api/api.client';
import { PageResponse } from '../../core/api/api.types';
import { RoomTypeCreateRequest, RoomTypeUpdateRequest } from '../../interfaces/request/room-type.request';
import { RoomTypeResponse } from '../../interfaces/response/room-type.response';

@Injectable({ providedIn: 'root' })
export class RoomTypesService {
  constructor(private api: ApiClient) {}

  listByHotel(hotelId: number, page: number, size: number) {
    return this.api.get<PageResponse<RoomTypeResponse>>('/api/room-types', { hotelId, page, size });
  }

  getById(id: number) {
    return this.api.get<RoomTypeResponse>(`/api/room-types/${id}`);
  }

  create(req: RoomTypeCreateRequest) {
    return this.api.post<RoomTypeResponse>('/api/room-types', req);
  }

  update(id: number, req: RoomTypeUpdateRequest) {
    return this.api.put<RoomTypeResponse>(`/api/room-types/${id}`, req);
  }

  delete(id: number) {
    return this.api.delete(`/api/room-types/${id}`);
  }
}

File: backend/src/main/java/com/fdsa/altairis/service/RoomTypesService.java
package com.fdsa.altairis.service;

import com.fdsa.altairis.dto.request.RoomTypeRequest;
import com.fdsa.altairis.dto.response.RoomTypeResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface RoomTypesService {
    RoomTypeResponse create(RoomTypeRequest request, String user);
    RoomTypeResponse update(Long id, RoomTypeRequest request, String user);
    RoomTypeResponse delete(Long id, String user);
    RoomTypeResponse getById(Long id);
    Page<RoomTypeResponse> listByHotel(Long hotelId, Pageable pageable);
}
